src/main/java/mate/academy/springboot/datajpa/controllers/ProductControllers.java

```diff
@@ -0,0 +1,102 @@
+package mate.academy.springboot.datajpa.controllers;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import mate.academy.springboot.datajpa.dto.mapper.CategoryMapper;
+import mate.academy.springboot.datajpa.dto.mapper.ProductMapper;
+import mate.academy.springboot.datajpa.dto.request.ProductRequestDto;
+import mate.academy.springboot.datajpa.dto.response.ProductResponseDto;
+import mate.academy.springboot.datajpa.models.Product;
+import mate.academy.springboot.datajpa.services.CategoryService;
+import mate.academy.springboot.datajpa.services.ProductService;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/products")
+public class ProductControllers {
+    private final ProductService productService;
+    private final ProductMapper productMapper;
+    private final CategoryMapper categoryMapper;
+    private final CategoryService categoryService;
+
+    public ProductControllers(ProductService productService,
+                              ProductMapper mapper,
+                              CategoryMapper categoryMapper,
+                              CategoryService categoryService) {
+        this.productService = productService;
+        this.productMapper = mapper;
+        this.categoryMapper = categoryMapper;
+        this.categoryService = categoryService;
+    }
+
+    @GetMapping("/inject")

```

Comment:
move to the Injector

Reply:
I forgot to delete these data, it were necessary for cheking workability of application).